ropensci/DataPackageR
# Improve and streamline internal functions handling DESCRIPTION and DataVersion

- new internal function validate_DataVersion():
  - does sanity checks on input DataVersion string that we want, like ensuring it is not NULL, is not NA, is length 1, and is of class 'character' or 'package_version'
  - uses base::package_version() to convert to class 'package_version', which performs additional sanity checks on the version (e.g., no letters)
  - check that it is a proper 3-part DataVersion (e.g., not '1.0')
  - converts back to class 'character' on output
  - now used in various other internal functions as sanity check

- new internal function validate_description():
  - uses desc::desc() to verify that there is a valid DESCRIPTION
  - also calls validate_DataVersion(), verifying the additional requirement that the package DESCRIPTION contains a valid DataVersion

- modify internal function validate_package_skeleton():
  - now uses desc::desc() to verify that there is a valid DESCRIPTION

- remove two internal functions, read.description() and read_pkg_description()
  - The former was an old non-exported roxygen2:: function that was copied over into DataPackageR after it was removed from roxygen2, and the latter wrapped the former
  - Unnecessary now that we are using desc::desc() and class 'description' to handle DESCRIPTION files

- modify internal function .increment_data_version():
  - its first argument and the first element of its return value is now an object of class 'description', e.g. from desc::desc(), rather than a custom object
  - now uses base R class 'package_version' and related functions, instead of former more complex code, to handle version subsetting, incrementing, and valid formatting
  - calls validate_DataVersion() internally before and after conversion to class 'package_version'

- modify internal function .digest_data_env():
  - its 3rd argument is now just the DataVersion string instead of more complex custom object containing various info from the package description
  - calls validate_DataVersion() internally to check DataVersion input

- modify internal function .save_data():
  - its 1st argument is now just the DataVersion string instead of more complex custom object containing various info from the package description
  - calls validate_DataVersion() internally to check input

- update a few tests, including adding several tests for validate_DataVersion()

- remove dependency on stringr

- No changes to UI or user-facing behaviors

- Passes all automated checks and tests locally; also tested interactively with some real-world data packages
